framework/src/packets/ip/v6/srh.rs

```diff
@@ -385,9 +399,36 @@ impl<E: Ipv6Packet> IpPacket for SegmentRouting<E> {
         self.envelope_mut().set_dst(dst)
     }
 
+    // From https://tools.ietf.org/html/rfc8200#section-8.1
+    //
+    // If the IPv6 packet contains a Routing header, the Destination Address
+    // used in the pseudo-header is that of the final destination.  At the
+    // originating node, that address will be in the last element of the Routing
+    // header; at the recipient(s), that address will be in the Destination
+    // Address field of the IPv6 header.
     #[inline]
-    fn pseudo_header_sum(&self, packet_len: u16, protocol: ProtocolNumber) -> u16 {
-        self.envelope().pseudo_header_sum(packet_len, protocol)
+    fn pseudo_header(&self, packet_len: u16, protocol: ProtocolNumber) -> PseudoHeader {
+        let dst = if self.segments_left() > 0 {
+            let segments = self.segments();
+            segments[segments.len() - 1]
+        } else {
+            match self.dst() {
```

Comment:
@drunkirishcoder wonder if you think we should make this match (and for src) it's own fun since we may need to convert in other places? Maybe rule of 3 times policy?

Reply:
a little annoying but it's not that much code.

Reply:
yep. we'll see if we hit this more times than not.